modules/icgc-argo-workflows/payload/alignment/resources/usr/bin/main.py

```diff
@@ -0,0 +1,202 @@
+    os.symlink(os.path.abspath(f), dst)
+
+    return dst
+
+
+def get_files_info(file_to_upload):
+    return {
+        'fileName': os.path.basename(file_to_upload),
+        'fileType': file_to_upload.split(".")[-1].upper(),
+        'fileSize': calculate_size(file_to_upload),
+        'fileMd5sum': calculate_md5(file_to_upload),
+        'fileAccess': 'controlled',
+        'dataType': 'Aligned Reads' if file_to_upload.split(".")[-1] in ('bam', 'cram') else 'Aligned Reads Index',
+        'info': {
+            'data_category': 'Sequencing Reads',
+            'data_subtypes': None
+            }
+    }
+
+
+def get_sample_info(sample_list):
+    samples = copy.deepcopy(sample_list)
+    for sample in samples:
+        for item in ['info', 'sampleId', 'specimenId', 'donorId', 'studyId']:
+            sample.pop(item, None)
+            sample['specimen'].pop(item, None)
+            sample['donor'].pop(item, None)
+
+    return samples
+
+
+def main(args):
+    with open(args.seq_experiment_analysis, 'r') as f:
+        seq_experiment_analysis_dict = json.load(f)
+
+    pipeline_info = {}
+    if args.pipeline_yml:
+        with open(args.pipeline_yml, 'r') as f:
+            pipeline_info = yaml.safe_load(f)
+
+    aligner = ""
+    updated_pipeline_info = {}
+    for key, value in pipeline_info.items():
+        new_key = key.split(":")[-1]
+        updated_pipeline_info[new_key] = value
+        if new_key == 'HISAT2_ALIGN':
+            aligner = "hisat2"
+        if new_key == "STAR_ALIGN":
+            aligner = "star"
+        if new_key == "BWA_MEM":
+            aligner = "bwa"
+        if new_key == "BWA_MEM2":
+            aligner = "bwa2"
+
+    for key, value in updated_pipeline_info.items():
+        for sub_key, sub_value in value.items():
+            value[sub_key] = str(sub_value)
+        updated_pipeline_info[key] = value
+
+    payload = {
+        'analysisType': {
+            'name': 'sequencing_alignment'
+        },
+        'studyId': seq_experiment_analysis_dict.get('studyId'),
+        'workflow': {
+            'workflow_name': args.wf_name,
+            'workflow_version': args.wf_version,
+            'genome_build': args.genome_build,
+            'run_id': args.wf_run,
+            'session_id': args.wf_session,
+            'pipeline_info': updated_pipeline_info,
+            'inputs': [
+                {
+                    'analysis_type': 'sequencing_experiment',
+                    'input_analysis_id': seq_experiment_analysis_dict.get('analysisId')
+                }
+            ]
+        },
+        'files': [],
+        'samples': get_sample_info(seq_experiment_analysis_dict.get('samples')),
+        'experiment': {},
+        'read_group_count': seq_experiment_analysis_dict.get('read_group_count'),
+        'read_groups': seq_experiment_analysis_dict.get('read_groups')
+    }
+    if args.genome_annotation != "[]":
+        payload['workflow']['genome_annotation'] = args.genome_annotation
```

Comment:
updated code to handle `genome_annotation` correctly. DNA alignment payload won't have `['workflow']['genome_annotation']`